x-pack/plugins/lens/public/drag_drop/providers.tsx

```diff
@@ -84,3 +92,84 @@ export function ChildDragDropProvider({ dragging, setDragging, children }: Provi
   const value = useMemo(() => ({ dragging, setDragging }), [setDragging, dragging]);
   return <DragContext.Provider value={value}>{children}</DragContext.Provider>;
 }
+
+interface ReorderState {
+  /**
+   * Ids of the elements that are translated up or down
+   */
+  reorderedItems: string[];
+
+  /**
+   * Direction of the move of dragged element in the reordered list
+   */
+  direction: '-' | '+';
+  /**
+   * height of the dragged element
+   */
+  draggingHeight: number;
+  /**
+   * indicates that user is in keyboard mode
+   */
+  isKeyboardReorderOn: boolean;
+  /**
+   * aria-live message for changes in reordering
+   */
+  keyboardReorderMessage: string;
+  /**
+   * reorder group needed for screen reader aria-described-by attribute
+   */
+  groupId: string;
+}
+
+export interface ReorderContextState {
+  reorderState: ReorderState;
+  setReorderState: (reorderState: ReorderState) => void;
+}
+
+export const ReorderContext = React.createContext<ReorderContextState>({
+  reorderState: {
+    reorderedItems: [],
+    direction: '-',
+    draggingHeight: 40,
+    isKeyboardReorderOn: false,
+    keyboardReorderMessage: '',
+    groupId: '',
+  },
+  setReorderState: () => {},
+});
+
+export function ReorderProvider({ id, children }: { id: string; children: React.ReactNode }) {
+  const [state, setState] = useState<ReorderContextState['reorderState']>({
+    reorderedItems: [],
+    direction: '-',
+    draggingHeight: 40,
+    isKeyboardReorderOn: false,
+    keyboardReorderMessage: '',
+    groupId: id,
+  });
+  const setReorderState = useMemo(() => (reorderState: ReorderState) => setState(reorderState), [
+    setState,
+  ]);
+
+  return (
+    <>
+      <ReorderContext.Provider value={{ reorderState: state, setReorderState }}>
+        {children}
+      </ReorderContext.Provider>
+      <EuiPortal>
+        <EuiScreenReaderOnly>
+          <>
+            <p id="lnsDragDrop-reorderAnnouncement" aria-live="assertive">
+              {state.keyboardReorderMessage}
+            </p>
```

Comment:
What happens if this message is empty? I didn't notice anything when testing with screen reader, just wondering.

Reply:
Just like EUI, there's nothing displayed.